### PR TITLE
fix logging for release

### PR DESCRIPTION
**Pull request by GlynLeine**

### CODE REVIEW

**Review by Ragingram2 — APPROVED** [review]

File: legion/engine/core/logging/logging.cpp [hunk 1]
```diff
@@ -2,9 +2,12 @@
 
 
 namespace legion::core::log {
-
-std::shared_ptr<spdlog::logger> impl::file_logger  = spdlog::rotating_logger_mt(LOG_FILE,LOG_FILE,1'048'576,5);
-std::shared_ptr<spdlog::logger> impl::console_logger = spdlog::stdout_color_mt("does-not-matter");
-std::unordered_map<std::thread::id, std::string> impl::thread_names;
+    cstring impl::log_file = nullptr;
+    cstring impl::file_file = "logs/legion-engine.log";
```
Comment (Algo-ryth-mix) [inline legion/engine/core/logging/logging.cpp]: `file_file` ?

File: legion/engine/core/logging/logging.hpp [hunk 2]
```diff
@@ -292,22 +297,21 @@ namespace legion::core::log
...
-
-#endif
 
     /** @brief sets up logging (do not call, invoked by engine) */
     inline void setup()
     {
+#if defined(LEGION_KEEP_CONSOLE) || defined(LEGION_DEBUG)
+        LOG_FILE = impl::console_file;
+        logger = impl::console_logger;
+#else
```
Comment (Algo-ryth-mix) [inline legion/engine/core/logging/logging.hpp]: I am not 100% certain that LOG_FILE is required at all when logging via console

Comment (Algo-ryth-mix) [inline legion/engine/core/logging/logging.hpp]: this is very likely to cause problems when events are logged before the engine is started up. I would recommend settting the default logger to be the console logger at first and replacing it with the file logger if necessary

**Review by Algo-ryth-mix — CHANGES_REQUESTED** [review]

**Comment by GlynLeine** [comment]
@Algo-ryth-mix I applied your requested changes!

**Review by Algo-ryth-mix — APPROVED** [review]

**Review by Algo-ryth-mix — APPROVED** [review]

**Pull request merged**